src/execution/__tests__/executor-test.js

```diff
@@ -306,11 +307,65 @@ describe('Execute: Handles basic execution tasks', () => {
     const field = operation.selectionSet.selections[0];
     expect(resolvedInfo).to.deep.include({
       fieldNodes: [field],
-      path: { prev: undefined, key: 'result' },
+      path: { prev: undefined, key: 'result', parentType: 'Test' },
       variableValues: { var: 'abc' },
     });
   });
 
+  it('populates path correctly with complex types', () => {
+    let path;
+    const a = new GraphQLObjectType({
+      name: 'A',
+      fields: {
+        test: {
+          type: GraphQLString,
+          resolve(_val, _args, _ctx, info) {
+            path = info.path;
+          },
+        },
+      },
+    });
+    const testUnion = new GraphQLUnionType({
+      name: 'TestUnion',
+      types: [a],
+      resolveType() {
+        return a;
```

Comment:
Just do 
```suggestion
        return 'A';
```

Reply:
I will, but could you explain why? To me it seems that it'd be more expensive, because now you have to look up the type by name?